Rakefile

```diff
@@ -114,10 +115,105 @@ namespace :arm do
+    subdir_list.each do |subdir|
+      if subdir != 'common' && subdir != 'azure_sdk'
+        folder_to_be_cleaned = File.expand_path("../azure_sdk/lib/#{subdir}", __FILE__)
+        puts "Cleaning folder - #{folder_to_be_cleaned}"
+        FileUtils.rm_rf(folder_to_be_cleaned)
+      end
+    end
+  end
+
+  desc 'Clean Individual Profiles'
+  task :clean_individual_profiles do
+    Dir.chdir(File.expand_path('../management', __FILE__))
+    gem_folders = Dir['*'].reject{|o| not File.directory?(o)}
+    gem_folders.each do |gem|
+      Dir.chdir(File.expand_path("../management/#{gem}/lib/profiles", __FILE__))
+      subdir_list = Dir['*'].reject{|o| not File.directory?(o)}
+      subdir_list.each do |subdir|
+        if subdir.to_s != 'common'
+          folder_to_be_cleaned = File.expand_path("../management/#{gem}/lib/profiles/#{subdir}", __FILE__)
+          puts "Cleaning folder - #{folder_to_be_cleaned}"
+          FileUtils.rm_rf(folder_to_be_cleaned)
+        end
+      end
+    end
+  end
+
+  desc 'Clean All profiles'
+  task :clean_all_profiles => [:clean_rollup_profiles, :clean_individual_profiles] do
+    puts 'Cleaned all profiles'
+  end
+
+  desc 'Regen rollup profiles'
+  task :regen_rollup_profile => :clean_rollup_profiles do
+    Dir.chdir(File.expand_path('../generators/profilegen/src', __FILE__))
+    command = "#{get_base_profile_generation_cmd}#{get_profile_spec_files_folder}/#{PROFILE_METADATA[:azure_sdk]}"
+    execute_and_stream(command)
+  end
+
+  desc 'Regen individual profiles'
+  task :regen_individual_profiles => :clean_individual_profiles do
+    Dir.chdir(File.expand_path('../generators/profilegen/src', __FILE__))
+    PROFILE_METADATA.each do |sdk, profile_spec_file|
+      if(sdk.to_s != 'azure_sdk')
+        command = "#{get_base_profile_generation_cmd}#{get_profile_spec_files_folder}/#{profile_spec_file}"
+        execute_and_stream(command)
+      end
+    end
+  end
+
+  desc 'Generate individual require files'
+  task :regen_individual_require_files do
+    Dir.chdir(File.expand_path('../generators/requirefilegen/src', __FILE__))
+    command = 'bundle exec ruby require_file_generator.rb --mode=individual'
+    execute_and_stream(command)
+  end
+
+  desc 'Generate rollup require files'
+  task :regen_rollup_require_files do
+    Dir.chdir(File.expand_path('../generators/requirefilegen/src', __FILE__))
+    command = 'bundle exec ruby require_file_generator.rb --mode=rollup'
+    execute_and_stream(command)
+  end
+
+  desc 'Generate all require files'
+  task :regen_all_require_files => [:regen_individual_require_files, :regen_rollup_require_files] do
+    puts 'Generated all require files'
+  end
+
+  desc 'Regen all profiles'
+  task :regen_all_profiles => [:regen_rollup_profile, :regen_individual_profiles] do
+    puts 'Regenerated all profiles'
+  end
+end
+
+Rake::Task['arm:regen_rollup_profile'].enhance do
+  Rake::Task['arm:regen_rollup_require_files'].invoke
```

Comment:
Curious: why do you need to explicitly invoke? We have built other tasks where we do not do that so I was curious.

Reply:
Synched up offline. Also added an enhancement on the regen task